logic.js

```diff
@@ -0,0 +1,215 @@
+let posts = [
+  {
+    text: 'First post!',
+    id: 'p1',
+    comments: [
+      { id: 'c1', text: 'First comment on first post!' },
+      { id: 'c2', text: 'Second comment on first post!!' },
+      { id: 'c3', text: 'Third comment on first post!!!' }
+    ]
+  },
+  {
+    text: 'Aw man, I wanted to be first',
+    id: 'p2',
+    comments: [
+      { id: 'c4', text: "Don't wory second poster, you'll be first one day." },
+      { id: 'c5', text: 'Yeah, believe in yourself!' },
+      { id: 'c6', text: 'Haha second place what a joke.' }
+    ]
+  }
+]
+
+const newLocal = 'tweetweData'
+if (!localStorage.getItem(newLocal)) {
+  saveData()
+} else {
+  posts = JSON.parse(localStorage.getItem(newLocal))
+}
+
+
+function saveData () {
+  localStorage.setItem(newLocal, JSON.stringify(posts))
+}
+
+let postIdCounter = posts.length + 1
+
+let commentIdCounter = 0
+for (let i = 0; i < posts.length; i++) {
+  commentIdCounter += posts[i].comments.length
+}
+
+function getPosts () {
+  return posts
+}
+
+function addPost (text) {
+  const newPost = {}
+  newPost.text = text
+  newPost.id = `p${postIdCounter}`
+  newPost.comments = []
+  posts.push(newPost)
+  postIdCounter++
+  saveData()
+  return posts
+}
+
+function removePost (postID) {
+  for (let i = 0; i < posts.length; i++) {
+    if (posts[i].id === postID) {
+      posts.splice(i, 1)
+    }
+  }
+  saveData()
+  return posts
```

Comment:
The removePost method should not return anything, but validation if the deleting where succeed)